unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/RPCSocialApiBridge.cs

```diff
@@ -210,12 +278,128 @@ public async UniTask<FriendRequest> RequestFriendshipAsync(string friendUserId,
                         Message = messageBody,
                         User = new User() { Address = friendUserId },
                     }
+                },
+                AuthToken = new Payload()
+                {
+                    SynapseToken = accessToken
                 }
             };
 
-            await UpdateFriendship(updateFriendshipPayload, cancellationToken);
+            return await UpdateFriendship(updateFriendshipPayload, friendUserId, cancellationToken);
+        }
+
+        private async UniTask SubscribeToFriendshipEvents(CancellationToken cancellationToken = default)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            try
+            {
+                await foreach (var friendshipEventResponse in socialClient.SubscribeFriendshipEventsUpdates(new Payload() { SynapseToken = accessToken }))
+                {
+                    switch (friendshipEventResponse.ResponseCase)
+                    {
+                        case SubscribeFriendshipEventsUpdatesResponse.ResponseOneofCase.InternalServerError:
+                        case SubscribeFriendshipEventsUpdatesResponse.ResponseOneofCase.UnauthorizedError:
+                        case SubscribeFriendshipEventsUpdatesResponse.ResponseOneofCase.ForbiddenError:
+                        case SubscribeFriendshipEventsUpdatesResponse.ResponseOneofCase.TooManyRequestsError:
+                            ProcessAndLogSubscriptionError(friendshipEventResponse);
+                            break;
+                        case SubscribeFriendshipEventsUpdatesResponse.ResponseOneofCase.Events:
+                            await ProcessFriendshipEventResponse(friendshipEventResponse.Events, cancellationToken);
+                            break;
+                        default:
+                        {
+                            Debug.LogErrorFormat("Subscription to friendship events got invalid response {0}", friendshipEventResponse);
+                            throw new ArgumentOutOfRangeException();
```

Comment:
Are you sure you want to break the whole loop if one event outbreaks?

Reply:
We dont want to break the loop, fixed it